Clean/Utils/Vector.lean

```diff
@@ -140,18 +140,39 @@ theorem induct_push_push {motive : {n: ℕ} → Vector α n → Sort u}
 def finRange (n : ℕ) : Vector (Fin n) n :=
   ⟨ .mk (List.finRange n), List.length_finRange n ⟩
 
-def mapFinRange {n} (create: Fin n → α) : Vector α n := finRange n |>.map create
+theorem finRange_zero : finRange 0 = #v[] := rfl
+
+theorem getElemFin_finRange {n} (i : Fin n) : (finRange n)[i] = i := by
+  simp [finRange, List.finRange]
+
+theorem getElem_finRange {n} (i : ℕ) (hi : i < n) : (finRange n)[i] = ⟨ i, hi ⟩ := by
+  simp [finRange, List.finRange]
+
+theorem finRange_succ {n} : finRange (n + 1) = ((finRange n).map Fin.castSucc).push n := by
+  ext i hi
+  simp only [getElem_finRange, getElem_push, getElem_map, Fin.castSucc_mk]
+  by_cases hi' : i < n <;> simp [hi']; linarith
+
+def mapFinRange (n: ℕ) (create: Fin n → α) : Vector α n := finRange n |>.map create
```

Comment:
`Vector.mapFinRange` is changed to take the length argument explicitly, for consistency with similar methods